src/components/reusable/table/table-header.ts

```diff
@@ -68,11 +96,34 @@ export class TableHeader extends LitElement {
     // Dispatch event to notify parent components of the sorting change
     this.dispatchEvent(
       new CustomEvent('on-sort-changed', {
-        detail: { sortDirection: this.sortDirection },
+        bubbles: true,
+        composed: true,
+        detail: { sortDirection: this.sortDirection, sortKey: this.sortKey },
       })
     );
   }
 
+  override updated() {
+    this.checkIfHeaderSlotIsEmpty();
+  }
+
+  checkIfHeaderSlotIsEmpty() {
+    // Retrieve the slot from the shadow DOM.
+    const slot = this.shadowRoot!.querySelector('slot');
+
+    // Get all nodes assigned to the slot.
+    const nodes = slot!.assignedNodes({ flatten: true });
```

Comment:
It might be easier to use the Lit decorator for this: https://lit.dev/docs/api/decorators/#queryAssignedNodes